source/common/tracing/http_tracer_impl.cc

```diff
@@ -244,11 +243,37 @@ LightStepDriver::LightStepDriver(const Json::Object& config,
   });
 }
 
-SpanPtr LightStepDriver::startSpan(const std::string& operation_name, SystemTime start_time) {
-  lightstep::Span span = tls_.getTyped<TlsLightStepTracer>(tls_slot_).tracer_.StartSpan(
-      operation_name, {lightstep::StartTimestamp(start_time)});
+SpanPtr LightStepDriver::startSpan(Http::HeaderMap& request_headers,
+                                   const std::string& operation_name, SystemTime start_time) {
+  lightstep::Tracer& tracer = tls_.getTyped<TlsLightStepTracer>(tls_slot_).tracer_;
+  LightStepSpanPtr active_span;
+
+  if (request_headers.OtSpanContext()) {
+    // Extract downstream context from HTTP carrier.
+    std::string parent_context = Base64::decode(request_headers.OtSpanContext()->value().c_str());
```

Comment:
if header is invalid we'll not populate CarrierStruct and we'll get spans joined in a single trace by x-request-id, not a big deal